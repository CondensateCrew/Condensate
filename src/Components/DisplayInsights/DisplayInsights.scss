@import '../../css/index.scss';
@import '../../css/variables.scss';

.display-insights-section {
  margin-top: $medium-space;
  border-radius: $one-line-border-radius;
  box-shadow: $external-shadow;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  .display-insights-header-div {
    display: flex;
    width: 90%;
    h3 {
      width: 90%;
    }
    justify-content: space-between;
    align-items: center;
    margin: $small-space;
  }
  img {
    width: 2.25rem;
    padding: .5rem;
    border-radius: $one-line-border-radius;
    box-shadow: $external-shadow;
  }
  button {
    @extend %regular-btn;
  }
}

.display-insights-body-div {
  height: 30vh;
  box-shadow: $inner-shadow;
  border-radius: $one-line-border-radius;
  width: 90%;
  padding: $small-space;
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow:scroll;

  h2 {
    font-size: 1rem;
    border-radius: $one-line-border-radius;
    box-shadow: $external-shadow;
    padding: $medium-space;
  }
  li {
    width: 90%;
    padding: $small-space;
    border-bottom: 1px solid rgba(181, 200, 219, 0.5);
  }
}