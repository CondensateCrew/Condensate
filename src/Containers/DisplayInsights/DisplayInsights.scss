@import '../../css/index.scss';
@import '../../css/variables.scss';

.display-insights-section {
  @extend %one-line-block;
  padding: $one-line-padding-space;
  margin: $medium-space 0;
  header {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    h3 {
      font-weight: 400;
      width: 100%;
    }
    button {
      display: -webkit-flex;
      display: -ms-flex;
      display: flex;
      align-items: center;
      justify-content: center;
      padding: $extra-small-space;
      background: $primary-gradient;
      border-radius: $one-line-border-radius;
      box-shadow: $external-shadow;
      img {
        width: $big-text;
        &.up-btn {
          transform: rotate(180deg);
        }
      }
    }
  }
  &>button {
    @extend %regular-btn;
  }
}

.display-insights-body-div {
  background: $primary-gradient;
  box-shadow: $inner-shadow;
  border-radius: $one-line-border-radius;
  padding: $small-space;
  display: flex;
  flex-direction: column;
  margin-top: $medium-space;
  width: 100%; max-height: 40vh;
  overflow: scroll;
  h2 {
    @extend %one-line-block;
    font-size: $small-text;
    line-height: 1.4em;
    padding: $medium-space;
    margin-bottom: $tiny-space;
    width: 100%;
  }
  li {
    padding: $small-space 0;
    margin: 0 $small-space;
    border-bottom: 2px solid $dark-shadow-primary-transparent;
    text-align: left;
    -webkit-hyphens: auto;
    -moz-hyphens: auto;
    -ms-hyphens: auto;
    hyphens: auto;
    overflow-wrap: break-word;
    width: 90%;
    &:last-child {
      border-bottom: none;
    }
  }
}
