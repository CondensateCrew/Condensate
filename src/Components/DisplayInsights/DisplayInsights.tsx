import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { AppStore } from 'interfaces';
import './DisplayInsights.scss';
import down from 'assets/down.svg';

const DisplayInsights:React.FC = () => {
  const [ currentStep, setCurrentStep ] = useState<number>(0);
  const [ isClicked, setIsClicked ] = useState<boolean>(false);
  const insights = useSelector((store: AppStore) => store.insights);

  const responses = insights[currentStep].answers.map((answer:string, index: number) => {
    return <li key={index}>{answer}</li>
  });

  const showNextInsight = ():void => {
    if (currentStep > 1) {
      setCurrentStep(0)
    } else {
      setCurrentStep(currentStep + 1)
    }
  };

  const toggleBlock = () => setIsClicked(!isClicked);
  const upClass = (isClicked) ? 'up-btn' : '';

  return (
    <section className='display-insights-section'>
      <header>
        <h3>Insights</h3>
        <button onClick={toggleBlock}>
          <img src={down} alt='down-arrow' className={upClass} />
        </button>
      </header>
      { isClicked &&
        <ul className='display-insights-body-div'>
          <h2>{insights[currentStep].question.sentence}</h2>
          {responses}
        </ul>
      }
      { isClicked &&
        <button type='button' onClick={showNextInsight}>Next</button>}
    </section>
  )
}

export default DisplayInsights;
