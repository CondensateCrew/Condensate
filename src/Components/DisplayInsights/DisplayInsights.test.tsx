import React from 'react';
import DisplayInsights from './DisplayInsights';
import { shallow } from 'enzyme';

describe('DisplayInsights component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<DisplayInsights />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if block is opened', () => {
    const wrapper = shallow(<DisplayInsights />);
    wrapper.find('header>button').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
