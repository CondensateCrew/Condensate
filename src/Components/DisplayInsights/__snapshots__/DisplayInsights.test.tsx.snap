// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayInsights component should match the snapshot 1`] = `
<section
  className="display-insights-section"
>
  <header>
    <h3>
      Insights
    </h3>
    <button
      onClick={[Function]}
    >
      <img
        alt="down-arrow"
        className=""
        src="down.svg"
      />
    </button>
  </header>
</section>
`;

exports[`DisplayInsights component should match the snapshot if block is opened 1`] = `
<section
  className="display-insights-section"
>
  <header>
    <h3>
      Insights
    </h3>
    <button
      onClick={[Function]}
    >
      <img
        alt="down-arrow"
        className="up-btn"
        src="down.svg"
      />
    </button>
  </header>
  <ul
    className="display-insights-body-div"
  >
    <h2>
      A goose flies south
    </h2>
    <li>
      Why is the goose flying?
    </li>
    <li>
      How does a goose fly?
    </li>
    <li>
      Why not fly south?
    </li>
  </ul>
  <button
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</section>
`;
